### PR TITLE
Dynamo DB table creation

### PR DESCRIPTION
**Pull request by jeetpatgiri**
- added release service database schema
- added unit test to test registering release service with the grpc server
- added release service endpoint
- resolved requested changes
- fixed an error found during initializing dynamodb tables, checking error types prevents tables from being created, more details in Jira

deployed on a testing environment - Release v0.0.60-CIEDEV-2141.2
image

### CODE REVIEW

File: internal/release/services/release/release.go [hunk 1]
```diff
@@ -1,24 +1,77 @@
...
+	Schema *dynamodb.CreateTableInput
+}
+
+type ReleaseInfo struct {
+	Type             string       `json:"type" bson:"type" dynamodbav:"type"`
+	Name             string       `json:"name" bson:"name" dynamodbav:"name"`
+	Version          string       `json:"version" bson:"version" dynamodbav:"version"`
+	Date             string       `json:"date" bson:"date" dynamodbav:"date"`
+	Link             release.Link `json:"link" bson:"link" dynamodbav:"link"`
+	ShortDescription string       `json:"short_description" bson:"short_description" dynamodbav:"short_description"`
```
Comment (vmilovanovicc) [inline internal/release/services/release/release.go:37]: Please add the following:
```
CreatedOn  string   `json:"created_on" bson:"created_on" dynamodbav:"created_on"`
ModifiedOn string   `json:"modified_on" bson:"modified_on" dynamodbav:"modified_on"`
```

**Review by vmilovanovicc — CHANGES_REQUESTED** [review]

File: cmd/gateway.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ func init() {
 	gatewayCmd.Flags().StringVarP(&version_manager.VersionManagerEndpoint, "version-manager", "", version_manager.DefaultVersionManagerEndpoint, "GRPC Endpoint for Version Manager Service")
 	gatewayCmd.Flags().StringVarP(&storage.StorageServiceEndpoint, "storage", "", storage.DefaultStorageServiceDefaultEndpoint, "GRPC Endpoint for Storage Service")
 	gatewayCmd.Flags().StringVarP(&tag_manager.TagManagerEndpoint, "tag-manager", "", tag_manager.DefaultTagManagerEndpoint, "GRPC Endpoint for Tag Service")
-
+	//gatewayCmd.Flags().StringVarP(&release.ReleaseEndpoint, "tag", "", release.DefaultReleaseEndpoint, "GRPC Endpoint for Tag Service")
```
Comment (odelanno) [inline cmd/gateway.go:28]: why is it commented out?

File: internal/release/services/release/release.go [hunk 3]
```diff
@@ -1,24 +1,84 @@
...
 	releaseSvc "github.com/terrariumcloud/terrarium/internal/release/services"
+	"github.com/terrariumcloud/terrarium/internal/storage"
+	"github.com/terrariumcloud/terrarium/pkg/terrarium/release"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	DefaultReleaseTableName = "terrarium-module-releases"
```
Comment (odelanno) [inline internal/release/services/release/release.go:18]: The name of the table is wrong it should be terrarium-releases as we are going to use release for many things.

File: internal/release/services/release/release.go [hunk 4]
```diff
@@ -1,24 +1,84 @@
...
+	ReleaseTableInitializationError = status.Error(codes.Unknown, "Failed to initialize table for releases.")
+	//ReleaseEndpoint                 = DefaultReleaseEndpoint
 )
 
 type ReleaseService struct {
 	releaseSvc.UnimplementedPublisherServer
-	// Db     storage.DynamoDBTableCreator
-	// Table  string
-	// Schema *dynamodb.CreateTableInput
+	Db     storage.DynamoDBTableCreator
```
Comment (odelanno) [inline internal/release/services/release/release.go:30]: indentation seems off by 1 character as it should be aligned after the *

File: internal/release/services/release/release.go [hunk 5]
```diff
@@ -1,24 +1,84 @@
...
+
+type ReleaseInfo struct {
+	Type             string       `json:"type" bson:"type" dynamodbav:"type"`
+	Name             string       `json:"name" bson:"name" dynamodbav:"name"`
+	Version          string       `json:"version" bson:"version" dynamodbav:"version"`
+	Date             string       `json:"date" bson:"date" dynamodbav:"date"`
+	Link             release.Link `json:"link" bson:"link" dynamodbav:"link"`
+	ShortDescription string       `json:"short_description" bson:"short_description" dynamodbav:"short_description"`
+	CreatedOn        string       `json:"created_on" bson:"created_on" dynamodbav:"created_on"`
+	ModifiedOn       string       `json:"modified_on" bson:"modified_on" dynamodbav:"modified_on"`
```
Comment (odelanno) [inline internal/release/services/release/release.go:43]: Why have a modified on, we don't support update.

File: internal/release/services/release/release.go [hunk 6]
```diff
@@ -1,24 +1,84 @@
...
+	Db     storage.DynamoDBTableCreator
+	Table  string
+	Schema *dynamodb.CreateTableInput
+}
+
+type ReleaseInfo struct {
+	Type             string       `json:"type" bson:"type" dynamodbav:"type"`
+	Name             string       `json:"name" bson:"name" dynamodbav:"name"`
+	Version          string       `json:"version" bson:"version" dynamodbav:"version"`
+	Date             string       `json:"date" bson:"date" dynamodbav:"date"`
```
Comment (odelanno) [inline internal/release/services/release/release.go:39]: Should be createdOn attributed why use a date.

**Review by odelanno — CHANGES_REQUESTED** [review]

File: cmd/gateway.go (hunk 2 above)
Comment (vmilovanovicc) [reply cmd/gateway.go:28]: Commented out for the time being, as it's not yet used in the gateway (or browse) service. We can uncomment. Thanks.

**Review by odelanno — DISMISSED** [review]

**Comment by ilijad1** [comment]
We need to add the semver package fix that @vmilovanovicc introduced to this PR.

**Review by ilijad1 — APPROVED** [review]
Tested and deployed on terrarium green cluster. Works as expected.

**Review by vmilovanovicc — APPROVED** [review]

**Pull request merged**